aws/resource_aws_api_gateway_usage_plan_key.go

```diff
@@ -15,6 +16,22 @@ func resourceAwsApiGatewayUsagePlanKey() *schema.Resource {
 		Create: resourceAwsApiGatewayUsagePlanKeyCreate,
 		Read:   resourceAwsApiGatewayUsagePlanKeyRead,
 		Delete: resourceAwsApiGatewayUsagePlanKeyDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				idParts := strings.Split(d.Id(), "/")
+				if len(idParts) != 3 || idParts[0] == "" || idParts[1] == "" || idParts[2] == "" {
+					return nil, fmt.Errorf("Unexpected format of ID (%q), expected USAGE-PLAN-ID/USAGE-PLAN-KEY-TYPE/USAGE-PLAN-KEY-ID", d.Id())
+				}
+				usagePlanId := idParts[0]
+				usagePlanKeyType := idParts[1]
+				usagePlanKeyId := idParts[2]
+				d.Set("usage_plan_id", usagePlanId)
+				d.Set("key_type", usagePlanKeyType)
+				d.Set("key_id", usagePlanKeyId)
+				d.SetId(usagePlanKeyId)
```

Comment:
Instead of requiring the key type during import, it seems we should instead fix the `Read` function to properly set the type, e.g.

```go
d.Set("key_type", up.Type)
```

Then remove the key type handling from the import function, testing, and documentation. 👍 

```go
				if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
					return nil, fmt.Errorf("Unexpected format of ID (%q), expected USAGE-PLAN-ID/USAGE-PLAN-KEY-ID", d.Id())
				}
				usagePlanId := idParts[0]
				usagePlanKeyId := idParts[1]
				d.Set("usage_plan_id", usagePlanId)
				d.Set("key_id", usagePlanKeyId)
				d.SetId(usagePlanKeyId)
```

Reply:
fixed